app/controllers/v6/assessments_controller.rb

```diff
@@ -48,7 +48,23 @@ def create
 
     def dependants(input_params, submission_date)
       dependant_params = input_params.fetch(:dependants, [])
-      dependant_params.map { |p| Dependant.new(p.merge(submission_date:)) }
+      dependant_params.map do |params|
+        updated_params = convert_monthly_income_params(params)
+        Dependant.new(updated_params.merge(submission_date:))
+      end
+    end
+
+    def convert_monthly_income_params(params)
+      income = params[:income]
+      monthly_income = params[:monthly_income]
+      if income
+        amount = income[:amount]
+        frequency = income[:frequency]
+      elsif monthly_income
+        amount = monthly_income
+        frequency = Dependant::DEFAULT_FREQUENCY
+      end
+      params.except!(:monthly_income, :income).merge!(amount:, frequency:)
```

Comment:
changing input parameters is usually a bad idea (and against our immutability ideas). params.except is fine here